docs/source/format_kg.rst

```diff
@@ -0,0 +1,215 @@
+Format of Input Data Used by DGL-KE
+-----------------------------------
+
+DGL-KE toolkits provide commands for training, evaluation and inference. Different commands require different kinds of input data, including:
+
+  * **Knowledge Graph** The knowledge graph used in train, evaluation and inference.
+  * **Trained Embeddings** The embedding generated by dglke_train or dglke_dist_train.
+  * **Other Data** Extra input data that used by inference tools.
+  
+A knowledge graph is usually stored in the form of triplets (head, relation, tail). Heads and tails are entities in the knowledge graph. All of them can be identified with unique IDs. In general, there exists two types of IDs for entities and relations in DGL-KE:
+
+  * **Raw ID** The entities and relations can be identified by names, usually in the format of strings.
+  * **KGE ID** They are used during knowledge graph training, evaluation and inference. Both entities and relations are identified with integers and should start from 0 and be contiguous.
+
+If the input file of a knowledge graph uses Raw IDs for entities and relations and does not provide a mapping between Raw IDs and KGE Ids, DGL-KE will generate an ID mapping automatically.
+
+The following table gives the overview of the input data for different toolkits. (Y for necessary and N for no-usage)
+
++------------------+----------+------------+---------------------+----------------+
+|  DGL-KE Toolkit  | Knowledge Graph       | Trained Embeddings  |   Other Data   |
++==================+==========+============+=====================+================+
+|                  | Triplets | ID Mapping |     Embeddings      |                |
++------------------+----------+------------+---------------------+----------------+
+| dglke_train      |    Y     |     N      |         N           |       N        |
```

Comment:
is it right to say that dgl_train doesn't require ID mapping? it supports inputs with ID mappings.

Reply:
Emm...
For UDD, the triplets are assumed already to be integers, But it check the ID mapping for integrity. 
For UDD_raw, the training script will generate the ID mapping.


ID mapping is useful in inference.